src/main/java/com/cleanroommc/groovyscript/api/IGameObjectHandler.java

```diff
@@ -45,13 +45,10 @@ static <T extends IForgeRegistryEntry<T>> IGameObjectHandler<T> wrapForgeRegistr
     static <T extends Enum<T>> IGameObjectHandler<T> wrapEnum(Class<T> enumClass, boolean caseSensitive) {
         Map<String, T> map = new Object2ObjectOpenHashMap<>();
         for (T t : enumClass.getEnumConstants()) {
-            map.put(t.name(), t);
-            if (caseSensitive) {
-                map.put(t.name().toUpperCase(Locale.ROOT), t);
-            }
+            map.put(caseSensitive ? t.name() : t.name().toUpperCase(Locale.ROOT), t);
         }
         return (s, args) -> {
-            T t = map.get(s);
+            T t = map.get(caseSensitive ? s : s.toUpperCase(Locale.ROOT));
```

Comment:
take an enum value of `thisExample`:
old, case sensitive: `thisExample` and `THISEXAMPLE` works
old, case insensitive: only `thisExample` works

new, case sensitive: only `thisExample` works
new, case insensitive: `thisExample`, `THISEXAMPLE`, and `tHiSeXaMpLe` works

so the reasoning is because
1. case sensitive logic was inverted (when true, added an UPPERCASE alt)
2. case insensitive was just case sensitive + UPPERCASE, instead of matching regardless of case.

Reply:
makes sense